### ISSUE TITLE
How to resolve {"level":"INFO","message":"[1:00:41.041 PM]: WARNING!!! using fail-open to connect. Driver is connecting to an HTTPS endpoint without OCSP based Certificated Revocation checking as it could not obtain a valid OCSP Response to use from the CA OCSP responder. Details:"}

### ISSUE THREAD

**Issue by markddrake** [snowflakedb/snowflake-connector-nodejs#371]
```
root@b08f26fcd019:/usr/src/YADAMU# node --version
v18.8.0

root@b08f26fcd019:/usr/src/YADAMU# npm --version
8.18.0

root@b08f26fcd019:/usr/src/YADAMU# lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 11 (bullseye)
Release:        11
Codename:       bullseye
root@b08f26fcd019:/usr/src/YADAMU#
```

After my program has been running for serveral  hours the following message appears in the log file.

```
2022-12-02T15:24:54.636Z [QA][Environemnt][x64][linux][v18.8.0]: Running tests
2022-12-02T15:24:56.020Z [INFO][SNOWFLAKE][6.39.0][Configuration]: XMLType storage model is TEXT.
2022-12-02T15:24:56.204Z [INFO][READER][YABASC][3.0][Manager]: Ready.
2022-12-02T15:24:56.207Z [INFO][WRITER][SNOWFLAKE][6.39.0][DATA_ONLY][Manager]: Ready.
2022-12-02T15:24:57.551Z [INFO][YABASC][READER]: Processing file "/usr/src/YADAMU/stagingArea/export/json/oracle/HR.json". Size 19754 bytes.
2022-12-02T15:24:58.895Z [DDL][SNOWFLAKE]: Generated 7 "Create Table" statements and 7 DML statements. Elapsed time: 00:00:00.023s.
2022-12-02T15:24:59.341Z [DDL][SNOWFLAKE]: Executed 7 DDL operations. Elapsed time: 00:00:00.445s.
2022-12-02T15:25:00.568Z [INFO][JOBS][Batch]: Rows 19. Reader Elapsed Time: 00:00:00.000s. Throughput 25942 rows/s. Writer Elapsed Time: 00:00:02.985s.  Idle Time: 00:00:00.000s. SQL Exection Time: 00:00:00.000s. Throughput: 6 rows/s.
2022-12-02T15:25:01.784Z [INFO][REGIONS][Batch]: Rows 4. Reader Elapsed Time: 00:00:00.000s. Throughput 45498 rows/s. Writer Elapsed Time: 00:00:01.211s.  Idle Time: 00:00:00.000s. SQL Exection Time: 00:00:00.000s. Throughput: 3 rows/s.
2022-12-02T15:25:03.145Z [INFO][COUNTRIES][Batch]: Rows 25. Reader Elapsed Time: 00:00:00.000s. Throughput 53304 rows/s. Writer Elapsed Time: 00:00:01.359s.  Idle Time: 00:00:00.000s. SQL Exection Time: 00:00:00.000s. Throughput: 18 rows/s.
2022-12-02T15:25:04.606Z [INFO][EMPLOYEES][Batch]: Rows 107. Reader Elapsed Time: 00:00:00.005s. Throughput 18262 rows/s. Writer Elapsed Time: 00:00:01.458s.  Idle Time: 00:00:00.000s. SQL Exection Time: 00:00:00.000s. Throughput: 73 rows/s.
2022-12-02T15:25:05.807Z [INFO][LOCATIONS][Batch]: Rows 23. Reader Elapsed Time: 00:00:00.000s. Throughput 28492 rows/s. Writer Elapsed Time: 00:00:01.199s.  Idle Time: 00:00:00.000s. SQL Exection Time: 00:00:00.000s. Throughput: 19 rows/s.
2022-12-02T15:25:07.008Z [INFO][DEPARTMENTS][Batch]: Rows 27. Reader Elapsed Time: 00:00:00.000s. Throughput 40409 rows/s. Writer Elapsed Time: 00:00:01.198s.  Idle Time: 00:00:00.000s. SQL Exection Time: 00:00:00.000s. Throughput: 23 rows/s.
2022-12-02T15:25:08.166Z [INFO][JOB_HISTORY][Batch]: Rows 10. Reader Elapsed Time: 00:00:00.000s. Throughput 32710 rows/s. Writer Elapsed Time: 00:00:01.157s.  Idle Time: 00:00:00.000s. SQL Exection Time: 00:00:00.000s. Throughput: 9 rows/s.
2022-12-02T15:25:08.177Z [INFO][YADAMU][TEST]: Operation completed successfully. Elapsed time: 00:00:13.536.
2022-12-02T15:25:09.244Z [INFO][SNOWFLAKE][6.39.0][Configuration]: XMLType storage model is TEXT.
...
2022-12-03T13:00:34.069Z [INFO][CUSTOMERS][OBJECT]: Rows committed: 50000.
{"level":"INFO","message":"[1:00:41.041 PM]: WARNING!!! using fail-open to connect. Driver is connecting to an HTTPS endpoint without OCSP based Certificated Revocation checking as it could not obtain a valid OCSP Response to use from the CA OCSP responder. Details:"}
2022-12-03T13:00:41.324Z [INFO][SALES][OBJECT]: Rows committed: 700000.
{"level":"INFO","message":"[1:00:41.041 PM]: WARNING!!! using fail-open to connect. Driver is connecting to an HTTPS endpoint without OCSP based Certificated Revocation checking as it could not obtain a valid OCSP Response to use from the CA OCSP responder. Details:"}
2022-12-03T13:00:41.324Z [INFO][SALES][OBJECT]: Rows committed: 700000.
2022-12-03T13:00:45.598Z [INFO][SALES][OBJECT]: Rows committed: 750000.
2022-12-03T13:00:49.753Z [INFO][SALES][OBJECT]: Rows committed: 800000.
2022-12-03T13:00:53.967Z [INFO][SALES][OBJECT]: Rows committed: 850000.
2022-12-03T13:00:58.437Z [INFO][SALES][OBJECT]: Rows committed: 900000.
2022-12-03T13:00:58.629Z [INFO][SALES][OBJECT]: Rows 918843. Reader Elapsed Time: 00:01:36.028s. Throughput 9568 rows/s. Writer Elapsed Time: 00:01:37.665s.  Idle Time: 00:00:00.000s. SQL Exection Time: 00:00:00.000s. Throughput: 9408 rows/s.
2022-12-03T13:00:58.633Z [INFO][YADAMU][TEST]: Operation completed successfully. Elapsed time: 00:01:42.404.
2022-12-03T13:00:58.633Z [QA][DBROUNDTRIP][COPY][snow#1://"ORACLE"."SH"][mongodb#1://"SH1"]: Errors: 0. Warnings: 0. Failed: 0. Elapsed Time: 00:01:42.403s.
2022-12-03T13:00:58.638Z [INFO][MongoDB][6.0.1][Configuration]: Document ID Tranformation: STRIP.
2022-12-03T13:00:58.638Z [INFO][MongoDB][6.0.1][Configuration]: Read Tranformation: DOCUMENT_TO_ARRAY.
2022-12-03T13:00:58.638Z [INFO][MongoDB][6.0.1][Configuration]: Write Tranformation: ARRAY_TO_DOCUMENT.
{"level":"INFO","message":"[1:00:58.058 PM]: WARNING!!! using fail-open to connect. Driver is connecting to an HTTPS endpoint without OCSP based Certificated Revocation checking as it could not obtain a valid OCSP Response to use from the CA OCSP responder. Details:"}
{"level":"INFO","message":"[1:00:58.058 PM]: WARNING!!! using fail-open to connect. Driver is connecting to an HTTPS endpoint without OCSP based Certificated Revocation checking as it could not obtain a valid OCSP Response to use from the CA OCSP responder. Details:"}
{"level":"INFO","message":"[1:00:58.058 PM]: WARNING!!! using fail-open to connect. Driver is connecting to an HTTPS endpoint without OCSP based Certificated Revocation checking as it could not obtain a valid OCSP Response to use from the CA OCSP responder. Details:"}
{"level":"INFO","message":"[1:00:58.058 PM]: WARNING!!! using fail-open to connect. Driver is connecting to an HTTPS endpoint without OCSP based Certificated Revocation checking as it could not obtain a valid OCSP Response to use from the CA OCSP responder. Details:"}
{"level":"INFO","message":"[1:00:59.059 PM]: WARNING!!! using fail-open to connect. Driver is connecting to an HTTPS endpoint without OCSP based Certificated Revocation checking as it could not obtain a valid OCSP Response to use from the CA OCSP responder. Details:"}
{"level":"INFO","message":"[1:01:00.10 PM]: WARNING!!! using fail-open to connect. Driver is connecting to an HTTPS endpoint without OCSP based Certificated Revocation checking as it could not obtain a valid OCSP Response to use from the CA OCSP responder. Details:"}
2022-12-03T13:01:00.243Z [INFO][SNOWFLAKE][6.39.0][Configuration]: XMLType storage model is TEXT.
{"level":"INFO","message":"[1:01:00.10 PM]: WARNING!!! using fail-open to connect. Driver is connecting to an HTTPS endpoint without OCSP based Certificated Revocation checking as it could not obtain a valid OCSP Response to use from the CA OCSP responder. Details:"}
....
 
The message does not appear to cause any problem in the running code (eg results are as execpted), however I do not appear to be able to prevent / trap / or suppress this message and it bulks up the log files.

**Comment by sfc-gh-dszmolka** [comment]
i could reproduce the issue as well. seems to be related how the OCSP local cache entries are only valid for a certain number of hours and after that period, they are considered expired. 
sadly the driver's current version isn't particularly helpful with detecting that (that the issue is, in fact, with the expired entries), as it doesn't (yet) log the actual error message; it's missing from after `Details: `. A fix incoming for that: #397 

Back to the original issue - we'll take a look why the entries are not refreshed and eventually lead to this warning, if program runs long enough. Since the default behaviour of the driver is [OCSP Fail-Open](https://docs.snowflake.com/en/user-guide/ocsp.html#fail-open) this is indeed just a warning in the logs and should not block the connection actually commencing.

**Comment by sfc-gh-dszmolka** [comment]
#442 should fix once merged and released

**Comment by sfc-gh-dszmolka** [comment]
fix released with [1.6.21](https://github.com/snowflakedb/snowflake-connector-nodejs/releases/tag/v1.6.21)
